test-suite/testkit/src/api.rs

```diff
@@ -360,6 +423,51 @@ where
 
         Err(error)
     }
+
+    fn response_to_api_result_new<R>(mut response: Response) -> api::ApiResult<R>
+    where
+        R: DeserializeOwned + 'static,
+    {
+        fn extract_error(response: &mut Response) -> Option<api::ApiError> {
```

Comment:
Probably it would make sense to implement `ApiError::parse(u16, AsRef<str>)` method and here just use 

```rust
ApiError::parse(code, body).expect("Failed to decode api error")
```